### PR TITLE
Hear formatting fix

### PR DESCRIPTION
**Pull request by MarcellusPye**
Fixes issue #7457 by cleaning the message of formatting before processing it.

TLDR Prevents unconscious people from hearing formatting

### CODE REVIEW

**Comment by Tayyyyyyy** [comment]
You may want to consider adding this to /code/__HELPERS/text.dm if it's not there already. Also the code base uses lowercase and underscores more often than it uses camel case.

File: code/modules/mob/hear_say.dm [hunk 1]
```diff
@@ -167,7 +167,22 @@
 
 /mob/proc/hear_sleep(var/message)
 	var/heard = ""
+	var/opentag = 0
+	var/closetag = 0
 	if(prob(15))
+		while(1) //Remove the HTML formatting so people don't hear colour
```
Comment (RemieRichards) [inline code/modules/mob/hear_say.dm]: there's a proc for removing html from text, it's santize_html iirc.

Comment (Tayyyyyyy) [inline code/modules/mob/hear_say.dm]: I don't think it's sanitize_html. It's either strip_html_simple or strip_html_properly.

Comment (RemieRichards) [inline code/modules/mob/hear_say.dm]: Yes, you're correct.

Comment (MarcellusPye) [inline code/modules/mob/hear_say.dm]: I know, I just couldn't find them because the wiki page is out of date and the GitHub search is case sensitive. Also /proc/strip_html_properly would be the correct usage but it's a bit too strict for this case.

Comment (MarcellusPye) [inline code/modules/mob/hear_say.dm]: Although I would have used it if I had found it.

Comment (Fox-McCloud) [inline code/modules/mob/hear_say.dm]: Could you please utilize it then, please?

Comment (MarcellusPye) [inline code/modules/mob/hear_say.dm]: strip_htmp_properly eliminates anything after an open tag or before a close tag, not just things inside tags, this could cause other odd behavior with custom scripts if I'm not mistaken. However I will change it if you prefer.

Comment (Fox-McCloud) [inline code/modules/mob/hear_say.dm]: After talking it over with @tigercat2000 and they're fairly sure `strip_html_properly` will work.

Comment (Tayyyyyyy) [inline code/modules/mob/hear_say.dm]: @MarcellusPye https://pastebin.com/raw/gHGi3Rhk if you need to test with script that has a decent amount of formatting.

Comment (MarcellusPye) [inline code/modules/mob/hear_say.dm]: It's some really minor edge cases that don't really matter.
